integrations/cognito/src/index.tsx

```diff
@@ -0,0 +1,357 @@
+import { sign } from '@tsndr/cloudflare-worker-jwt';
+import { Router } from 'itty-router';
+
+import { IntegrationInstallationConfiguration } from '@gitbook/api';
+import {
+    createIntegration,
+    FetchEventCallback,
+    Logger,
+    RuntimeContext,
+    RuntimeEnvironment,
+    createComponent,
+} from '@gitbook/runtime';
+
+const logger = Logger('cognito.visitor-auth');
+
+type CognitoRuntimeEnvironment = RuntimeEnvironment<
+    {},
+    CognitoSiteOrSpaceInstallationConfiguration
+>;
+
+type CognitoRuntimeContext = RuntimeContext<CognitoRuntimeEnvironment>;
+
+type CognitoSiteOrSpaceInstallationConfiguration = {
+    client_id?: string;
+    cognito_domain?: string;
+    client_secret?: string;
+};
+
+type CognitoState = CognitoSiteOrSpaceInstallationConfiguration;
+
+type CognitoProps = {
+    installation: {
+        configuration?: IntegrationInstallationConfiguration;
+    };
+    spaceInstallation: {
+        configuration?: CognitoSiteOrSpaceInstallationConfiguration;
+    };
+    siteInstallation: {
+        configuration?: CognitoSiteOrSpaceInstallationConfiguration;
+    };
+};
+
+export type CognitoAction = { action: 'save.config' };
+
+const getDomainWithHttps = (url: string): string => {
+    if (url.startsWith('https://') || url.startsWith('http://')) {
```

Comment:
Should you convert all traffic to `https://` ?

Reply:
Yes, there is no reason that the AWS Cognito user domain has the `http://` protocol. So we convert all domains without a protocol to `https://`.

but if the user, for some reason, specifies the `http://` protocol, we keep it. But it's unlikely to work. I am only keeping this here because of some edge case for the need of an `http://` protocol that I am unaware of (could not find anything in the Cognito docs)

Reply:
I don't think we should keep it. We should force all communication to be https